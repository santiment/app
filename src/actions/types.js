// app
export const SHOW_NOTIFICATION = '[app] SHOW_NOTIFICATION'
export const HIDE_NOTIFICATION = '[app] HIDE_NOTIFICATION'
export const APP_CHANGE_ONLINE_STATUS = '[app] CHANGE_ONLINE_STATUS'
export const APP_LAUNCHED = '[app] LAUNCHED'
export const APP_SHOW_ONBOARDING = '[app] SHOW_ONBOARDING'
export const APP_GDPR_FAILED = '[app] GDPR_FAILED'
export const APP_USER_HAS_INACTIVE_TOKEN = '[app] USER_HAS_INACTIVE_TOKEN'
export const APP_USER_NIGHT_MODE_SAVE = '[app] USER_NIGHT_MODE_SAVE'
export const APP_USER_BETA_MODE_SAVE = '[app] USER_BETA_MODE_SAVE'
export const APP_TOGGLE_SEARCH_FOCUS = '[app] TOGGLE_SEARCH_FOCUS'

// user
export const CHANGE_USER_DATA = '[user] CHANGE_USER_DATA'
export const USER_LOGIN_FAILED = '[user] LOGIN_FAILED'
export const USER_LOGIN_SUCCESS = '[user] LOGIN_SUCCESS'
export const USER_LOGIN_PENDING = '[user] LOGIN_PENDING'
export const USER_EMAIL_LOGIN = '[user] EMAIL_LOGIN'
export const USER_EMAIL_CHANGE = '[user] EMAIL_CHANGE'
export const USER_DIGEST_CHANGE = '[user] DIGEST_CHANGE'
export const USER_USERNAME_CHANGE = '[user] USERNAME_CHANGE'
export const USER_APIKEY_GENERATE = '[user] APIKEY_GENERATE'
export const USER_APIKEY_GENERATE_SUCCESS = '[user] APIKEY_GENERATE_SUCCESS'
export const USER_APIKEY_REVOKE = '[user] APIKEY_REVOKE'
export const USER_APIKEY_REVOKE_SUCCESS = '[user] APIKEY_REVOKE_SUCCESS'
export const USER_ETH_LOGIN = '[user] ETH_LOGIN'
export const USER_LOGOUT_SUCCESS = '[user] LOGOUT_SUCCESS'
export const USER_TOGGLE_PRIVACY_POLICY = '[user] TOGGLE_PRIVACY_POLICY'
export const USER_TOGGLE_MARKETING = '[user] TOGGLE_MARKETING'
export const USER_SETTING_GDPR = '[user] SETTING_SETTING_GDPR'
export const USER_TOGGLE_NIGHT_MODE = '[user] TOGGLE_NIGHT_MODE'
export const USER_TOGGLE_BETA_MODE = '[user] TOGGLE_BETA_MODE'
// user / settings
export const SETTINGS_REMOVE_CONNECTED_WALLET = '[user] REMOVE_CONNECTED_WALLET'
export const SETTINGS_REMOVE_CONNECTED_WALLET_SUCCESS =
  '[user] REMOVE_CONNECTED_WALLET_SUCCESS'
export const SETTINGS_REMOVE_CONNECTED_WALLET_FAILED =
  '[user] REMOVE_CONNECTED_WALLET_FAILED'
export const SETTINGS_CONNECT_NEW_WALLET = '[user] CONNECT NEW WALLET'
export const SETTINGS_CONNECT_NEW_WALLET_SUCCESS =
  '[user] CONNECT NEW WALLET SUCCESS'
export const SETTINGS_CONNECT_NEW_WALLET_FAILED =
  '[user] CONNECT NEW WALLET FAILED'
export const SETTINGS_GENERATE_TELEGRAM_DEEP_LINK =
  '[user] GENERATE_TELEGRAM_DEEP_LINK'
export const SETTINGS_REVOKE_TELEGRAM_DEEP_LINK =
  '[user] REVOKE_TELEGRAM_DEEP_LINK'
export const SETTINGS_REVOKE_TELEGRAM_DEEP_LINK_SUCCESS =
  '[user] SETTINGS_REVOKE_TELEGRAM_DEEP_LINK_SUCCESS'
export const SETTINGS_GENERATE_TELEGRAM_DEEP_LINK_SUCCESS =
  '[user] GENERATE_TELEGRAM_DEEP_LINK_SUCCESS'
export const SETTINGS_GENERATE_TELEGRAM_DEEP_LINK_FAILED =
  '[user] GENERATE_TELEGRAM_DEEP_LINK_FAILED'
export const SETTINGS_CONNECT_TELEGRAM = '[user] CONNECT_TELEGRAM'
export const SETTINGS_CONNECT_TELEGRAM_SUCCESS =
  '[user] CONNECT_TELEGRAM_SUCCESS'
export const SETTINGS_CONNECT_TELEGRAM_FAILED = '[user] CONNECT_TELEGRAM_FAILED'
export const SETTINGS_CONNECT_TELEGRAM_CANCEL = '[user] CONNECT_TELEGRAM_CANCEL'
export const SETTINGS_TOGGLE_NOTIFICATION_CHANNEL =
  '[user] TOGGLE_NOTIFICATION_CHANNEL'
export const SETTINGS_TOGGLE_NOTIFICATION_CHANNEL_SUCCESS =
  '[user] TOGGLE_NOTIFICATION_CHANNEL_SUCCESS'
export const SETTINGS_TOGGLE_NOTIFICATION_CHANNEL_FAILED =
  '[user] TOGGLE_NOTIFICATION_CHANNEL_FAILED'

// assets
export const USER_COPY_ASSET_LIST = '[user] COPY_ASSET_LIST'
export const USER_ADD_NEW_ASSET_LIST = '[user] ADD_NEW_ASSET_LIST'
export const USER_ADD_NEW_ASSET_LIST_SUCCESS =
  '[user] ADD_NEW_ASSET_LIST_SUCCESS'
export const USER_ADD_NEW_ASSET_LIST_FAILED = '[user] ADD_NEW_ASSET_LIST_FAILED'
export const USER_ADD_NEW_ASSET_LIST_CANCEL =
  '[user] ADD_NEW_ASSET_LIST_CANCELED'
export const USER_REMOVE_ASSET_LIST = '[user] REMOVE_ASSET_LIST'
export const USER_REMOVE_ASSET_LIST_SUCCESS = '[user] REMOVE_ASSET_LIST_SUCCESS'
export const USER_REMOVE_ASSET_LIST_FAILED = '[user] REMOVE_ASSET_LIST_FAILED'
export const USER_ADD_ASSET_TO_LIST = '[user] ADD_ASSET_TO_LIST'
export const USER_ADD_ASSET_TO_LIST_SUCCESS =
  '[user] USER_ADD_ASSET_TO_LIST_SUCCESS'
export const USER_ADD_ASSET_TO_LIST_FAILED =
  '[user] USER_ADD_ASSET_TO_LIST_FAILED'
export const USER_REMOVE_ASSET_FROM_LIST = '[user] REMOVE_ASSET_FROM_LIST'
export const USER_REMOVED_ASSET_FROM_LIST_SUCCESS =
  '[user] REMOVED_ASSET_FROM_LIST'
export const USER_REMOVED_ASSET_FROM_LIST_FAILED =
  '[user] REMOVED_ASSET_FROM_LIST'
export const USER_CHOOSE_ASSET_LIST = '[user] USER_CHOOSE_ASSET_LIST'
export const ASSETS_FETCH = '[assets] ASSETS_FETCH'
export const ASSETS_FETCH_SUCCESS = '[assets] ASSETS_FETCH_SUCCESS'
export const ASSETS_FETCH_FAILED = '[assets] ASSETS_FETCH_FAILED'
export const ASSETS_SET_MIN_VOLUME_FILTER =
  '[assets] ASSETS_SET_MIN_VOLUME_FILTER'

// watchlists
export const WATCHLIST_TOGGLE_CONFIRM_DELETE_MODAL =
  '[watchlists] TOGGLE_CONFIRM_DELETE_MODAL'
