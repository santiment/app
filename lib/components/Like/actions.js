export const INSIGHTS_LIKE = '[voting] INSIGHTS_LIKE';
export const INSIGHT_LIKE_FAIL = '[voting] INSIGHTS_LIKE_FAIL';
export const FEED_EVENT_LIKE = '[voting] FEED_EVENT_LIKE';
export const FEED_EVENT_LIKE_FAIL = '[voting] FEED_EVENT_LIKE_FAIL';