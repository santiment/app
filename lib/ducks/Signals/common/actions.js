function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

export const SIGNAL_CREATE = '[signal] CREATE_SIGNAL';
export const SIGNAL_CREATE_SUCCESS = '[signal] CREATE_SIGNAL_SUCCESS';
export const SIGNAL_CREATE_FAILED = '[signal] CREATE_SIGNAL_FAILED';
export const SIGNAL_UPDATE = '[signal] UPDATE_SIGNAL';
export const SIGNAL_UPDATE_SUCCESS = '[signal] UPDATE_SIGNAL_SUCCESS';
export const SIGNAL_UPDATE_FAILED = '[signal] UPDATE_SIGNAL_FAILED';
export const SIGNAL_FETCH_ALL_SUCCESS = '[signal] FETCH_ALL_SUCCESS'; // update signal

export const SIGNAL_TOGGLE_BY_ID = '[signal] TOGGLE_BY_ID';
export const SIGNAL_TOGGLE_SUCCESS = '[signal] TOGGLE_BY_ID_SUCCESS';
export const SIGNAL_TOGGLE_FAILED = '[signal] TOGGLE_BY_ID_FAILED';
export const SIGNAL_REMOVE_BY_ID = '[signal] REMOVE_BY_ID';
export const SIGNAL_REMOVE_BY_ID_SUCCESS = '[signal] REMOVE_BY_ID_SUCCESS';
export const SIGNAL_REMOVE_BY_ID_FAILED = '[signal] REMOVE_BY_ID_FAILED';
export const createTrigger = payload => mutateTrigger({
  payload
});
export const updateTrigger = payload => mutateTrigger({
  payload,
  isEdit: true
});

const mutateTrigger = ({
  payload,
  isEdit
}) => {
  return {
    type: isEdit ? SIGNAL_UPDATE : SIGNAL_CREATE,
    payload: _objectSpread(_objectSpread({}, payload), {}, {
      settings: JSON.stringify(payload.settings)
    })
  };
};

export const toggleTrigger = ({
  id,
  isActive
}) => ({
  type: SIGNAL_TOGGLE_BY_ID,
  payload: {
    id,
    isActive: !isActive
  }
});
export const removeTrigger = id => ({
  type: SIGNAL_REMOVE_BY_ID,
  payload: {
    id
  }
});