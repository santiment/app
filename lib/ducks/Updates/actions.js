// timeseries
export const NEW_APP_AVAILABLE = '[version] NEW_APP_AVAILABLE';
export const IS_LATEST_APP = '[version] IS_LATEST_APP';
export const newAppAvailable = () => ({
  type: NEW_APP_AVAILABLE
});
export const markAsLatestApp = () => ({
  type: IS_LATEST_APP
});