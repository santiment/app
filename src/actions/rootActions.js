import {
  SHOW_NOTIFICATION,
  HIDE_NOTIFICATION,
  APP_CHANGE_ONLINE_STATUS,
  USER_DIGEST_CHANGE,
  APP_LAUNCHED,
} from './types'

export const showNotification = (payload = { title: 'Empty message' }) => {
  const newPayload = typeof payload === 'string' ? { title: payload } : payload

  return {
    type: SHOW_NOTIFICATION,
    payload: {
      ...newPayload,
      id: Date.now(),
    },
  }
}

export const hideNotification = (id) => ({
  type: HIDE_NOTIFICATION,
  payload: {
    id,
  },
})

export const changeNetworkStatus = (newtworkStatus) => ({
  type: APP_CHANGE_ONLINE_STATUS,
  payload: {
    isOnline: newtworkStatus,
  },
})

export const changeDigestSubscription = (type = 'WEEKLY') => ({
  type: USER_DIGEST_CHANGE,
  payload: type,
})

export const launchApp = () => ({ type: APP_LAUNCHED })
