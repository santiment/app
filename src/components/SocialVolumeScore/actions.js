export const SOCIALVOLUME_DATA_FETCH = '[socialVolume] DATA_FETCH'
export const SOCIALVOLUME_DATA_FETCH_CANCEL = '[socialVolume] DATA_FETCH_CANCEL'
export const SOCIALVOLUME_DATA_FETCH_SUCCESS =
  '[socialVolume] DATA_FETCH_SUCCESS'
export const SOCIALVOLUME_DATA_FETCH_FAILED = '[socialVolume] DATA_FETCH_FAILED'
