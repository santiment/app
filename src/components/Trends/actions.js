// trends
export const TRENDS_HYPED_FETCH = '[trends] HYPED_FETCH'
export const TRENDS_HYPED_FETCH_SUCCESS = '[trends] HYPED_FETCH_SUCCESS'
export const TRENDS_HYPED_FETCH_FAILED = '[trends] HYPED_FETCH_FAILED'
export const TRENDS_HYPED_WORD_SELECTED = '[trends] HYPED_WORD_SELECTED'
// tickers + slugs
export const TRENDS_HYPED_FETCH_TICKERS_SLUGS =
  '[trends] HYPED_FETCH_TICKERS_SLUGS'
export const TRENDS_HYPED_FETCH_TICKERS_SLUGS_SUCCESS =
  '[trends] HYPED_FETCH_TICKERS_SLUGS_SUCCESS'
export const TRENDS_HYPED_FETCH_TICKERS_SLUGS_FAILED =
  '[trends] HYPED_FETCH_TICKERS_SLUGS_FAILED'

export const TREND_WORD_SCORE_CHANGE_FETCH =
  '[trends] TREND_WORD_SCORE_CHANGE_FETCH'
export const TREND_WORD_SCORE_CHANGE_FULFILLED =
  '[trends] WORD_SCORE_CHANGE_FULFILLED'
export const TREND_WORD_SCORE_CHANGE_FAILED =
  '[trends] WORD_SCORE_CHANGE_FAILED'

export const TREND_WORD_VOLUME_CHANGE_FETCH =
  '[trends] WORD_VOLUME_CHANGE_FETCH'
export const TREND_WORD_VOLUME_CHANGE_FULFILLED =
  '[trends] WORD_VOLUME_CHANGE_FULFILLED'
export const TREND_WORD_VOLUME_CHANGE_FAILED =
  '[trends] WORD_VOLUME_CHANGE_FAILED'

export const TRENDS_CONNECTED_WORDS_SUCCESS = '[trends] CONNECTED_WORDS_SUCCESS'
export const TRENDS_CONNECTED_WORDS_FAILED = '[trends] CONNECTED_WORDS_FAILED'
export const TRENDS_CONNECTED_WORDS_OPTIMIZATION_SUCCESS =
  '[trends] CONNECTED_WORDS_OPTIMIZATION_SUCCESS'

export const TRENDS_SELECTED_WORDS = `[trends] SELECTED_WORDS`
export const TRENDS_SELECTED_WORDS_CLEAR = `[trends] SELECTED_WORDS_CLEAR`
