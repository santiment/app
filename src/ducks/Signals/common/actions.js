export const SIGNAL_CREATE = '[signal] CREATE_SIGNAL'
export const SIGNAL_CREATE_SUCCESS = '[signal] CREATE_SIGNAL_SUCCESS'
export const SIGNAL_CREATE_FAILED = '[signal] CREATE_SIGNAL_FAILED'
export const SIGNAL_UPDATE = '[signal] UPDATE_SIGNAL'
export const SIGNAL_UPDATE_SUCCESS = '[signal] UPDATE_SIGNAL_SUCCESS'
export const SIGNAL_UPDATE_FAILED = '[signal] UPDATE_SIGNAL_FAILED'
export const SIGNAL_FETCH_ALL = '[signal] FETCH_ALL'
export const SIGNAL_FETCH_ALL_SUCCESS = '[signal] FETCH_ALL_SUCCESS'
export const SIGNAL_FETCH_ALL_ERROR = '[signal] FETCH_ALL_ERROR'
// update signal
export const SIGNAL_TOGGLE_BY_ID = '[signal] TOGGLE_BY_ID'
export const SIGNAL_TOGGLE_SUCCESS = '[signal] TOGGLE_BY_ID_SUCCESS'
export const SIGNAL_TOGGLE_FAILED = '[signal] TOGGLE_BY_ID_FAILED'
export const SIGNAL_REMOVE_BY_ID = '[signal] REMOVE_BY_ID'
export const SIGNAL_REMOVE_BY_ID_SUCCESS = '[signal] REMOVE_BY_ID_SUCCESS'
export const SIGNAL_REMOVE_BY_ID_FAILED = '[signal] REMOVE_BY_ID_FAILED'
// preview of history signal points
export const SIGNAL_FETCH_HISTORY_POINTS = '[signal] FETCH_HISTORY_POINTS'
export const SIGNAL_FETCH_HISTORY_POINTS_SUCCESS =
  '[signal] FETCH_HISTORY_POINTS_SUCCESS'
export const SIGNAL_FETCH_HISTORY_POINTS_FAILED =
  '[signal] FETCH_HISTORY_POINTS_FAILED'

export const createTrigger = payload => mutateTrigger({ payload })
export const updateTrigger = payload => mutateTrigger({ payload, isEdit: true })

const mutateTrigger = ({ payload, isEdit }) => {
  return {
    type: isEdit ? SIGNAL_UPDATE : SIGNAL_CREATE,
    payload: {
      ...payload,
      settings: JSON.stringify(payload.settings)
    }
  }
}

export const toggleTrigger = ({ id, isActive }) => ({
  type: SIGNAL_TOGGLE_BY_ID,
  payload: {
    id,
    isActive: !isActive
  }
})

export const removeTrigger = id => ({
  type: SIGNAL_REMOVE_BY_ID,
  payload: { id }
})

export const fetchHistorySignalPoints = ({ cooldown, settings }) => ({
  type: SIGNAL_FETCH_HISTORY_POINTS,
  payload: {
    settings,
    cooldown
  }
})
