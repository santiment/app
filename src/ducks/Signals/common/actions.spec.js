/* eslint-env jasmine */
import { createTrigger } from './actions'

describe('Signals actions', () => {
  it('createTrigger should work', () => {
    const resultAction = createTrigger({
      metric: 'price',
      target: 'santiment',
      percentThreshold: '5',
      cooldown: '1h',
      timeWindow: '24h',
      channels: ['Telegram'],
      title: 'Check',
      description: 'check'
    })
    expect(resultAction).toMatchSnapshot()
  })
})
