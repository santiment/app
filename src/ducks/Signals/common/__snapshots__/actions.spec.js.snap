// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals actions createTrigger should work 1`] = `
Object {
  "payload": Object {
    "channels": Array [
      "Telegram",
    ],
    "cooldown": "1h",
    "description": "check",
    "metric": "price",
    "percentThreshold": "5",
    "settings": undefined,
    "target": "santiment",
    "timeWindow": "24h",
    "title": "Check",
  },
  "type": "[signal] CREATE_SIGNAL",
}
`;
